### ISSUE TITLE
Unable to find Static Directory: /tmp/sources/static/ 

### ISSUE THREAD

**Issue by vitr** [ryansb/hugo-lambda#8]
I was reviewing my lamda's logs and found this error on every run, despite that I have this folder in the input bucket.

```
2015-08-13T07:14:05.757Z    dcb35260-418a-11e5-9793-8fb8a323d550    done downloading 
2015-08-13T07:14:05.782Z    dcb35260-418a-11e5-9793-8fb8a323d550    Running hugo 
2015-08-13T07:14:05.956Z    dcb35260-418a-11e5-9793-8fb8a323d550    hugo-stdout: INFO: 2015/08/13 Using config file: /tmp/sources/config.toml ERROR: 2015/08/13 Unable to find Static Directory: /tmp/sources/static/ 
```

**Comment by ryansb** [comment]
Hm, is there anything else in the logs that might give more context, or is that the only thing that's there?

**Comment by vitr** [comment]
sorry, here is the full log, if it helps

```
START RequestId: dcb35260-418a-11e5-9793-8fb8a323d550 
2015-08-13T07:14:01.328Z    dcb35260-418a-11e5-9793-8fb8a323d550    Reading options from event: { Records:  [ { eventVersion: '2.0', eventSource: 'aws:s3', awsRegion: 'us-east-1', eventTime: '2015-08-13T07:13:56.814Z', eventName: 'ObjectCreated:Put', userIdentity: { principalId: 'A3H012QG9ZB9Z4' }, requestParameters: { sourceIPAddress: '202.58.227.252' }, responseElements:  { 'x-amz-request-id': '98F440C0A1A87D27', 'x-amz-id-2': 'D5yOa82jHxMh1bTaT1y9fxT804O4FNAK4IYFMmrDj6Z8VqSfAk4AU2tB83Hm5Q51' }, s3:  { s3SchemaVersion: '1.0', configurationId: 'Kappa-HugoLambdaGenerate-notification', bucket:  { name: 'input.vitr.website', ownerIdentity: { principalId: 'A3H012QG9ZB9Z4' }, arn: 'arn:aws:s3:::input.vitr.website' }, object:  { key: 'themes/material-design/images/screenshot.png', size: 224256, eTag: '43fbe3774e9968440eabed850ff4a115', versionId: '.gL3GdKxCo0zwfMz0Svs1kl9kK5XdVHN', sequencer: '0055CC43B2B22A3E51' } } } ] } 
2015-08-13T07:14:05.757Z    dcb35260-418a-11e5-9793-8fb8a323d550    done downloading 
2015-08-13T07:14:05.782Z    dcb35260-418a-11e5-9793-8fb8a323d550    Running hugo 
2015-08-13T07:14:05.956Z    dcb35260-418a-11e5-9793-8fb8a323d550    hugo-stdout: INFO: 2015/08/13 Using config file: /tmp/sources/config.toml ERROR: 2015/08/13 Unable to find Static Directory: /tmp/sources/static/  
2015-08-13T07:14:06.374Z    dcb35260-418a-11e5-9793-8fb8a323d550    hugo-stdout: INFO: 2015/08/13 found taxonomies: map[string]string
{
    "tag": "tags",
    "category": "categories"
}
2015-08-13T07:14:07.214Z    dcb35260-418a-11e5-9793-8fb8a323d550    hugo-stdout: WARN: 2015/08/13 Unable to locate layout for 404 page: [404.html theme/404.html] 0 draft content 0 future content  6 pages created 13 paginator pages created 6 tags created 2 categories created in 1251 ms  
2015-08-13T07:14:07.257Z    dcb35260-418a-11e5-9793-8fb8a323d550    hugo exited with code: 0 
2015-08-13T07:14:10.656Z    dcb35260-418a-11e5-9793-8fb8a323d550    done uploading 
2015-08-13T07:14:10.657Z    dcb35260-418a-11e5-9793-8fb8a323d550    All methods in waterfall succeeded. 
END RequestId: dcb35260-418a-11e5-9793-8fb8a323d550 
REPORT RequestId: dcb35260-418a-11e5-9793-8fb8a323d550  Duration: 9383.84 ms    Billed Duration: 9400 ms Memory Size: 128 MB    Max Memory Used: 27 MB  
```

**Comment by ryansb** [comment]
Does your site still generate properly, or is this error causing your site to fail? 

I think this may be a benign error due to the way hugo-lambda only downloads site content (templates, themes, etc) and not static files when generating the site. Mostly, the reason is to save time and keep from executing for too long if your site has big images, or other files.

**Comment by vitr** [comment]
nope, haven't managed to run it on aws. nevertheless could run hugo locally and upload results to s3, that was my workaround)

**Comment by ryansb** [comment]
I'm sorry you're having trouble. What's odd is that the log you posted _seems_ like it succeeded with the `6 pages created 13 paginator pages created`

Can you try emptying out your output bucket and using kappa to invoke the function? Here are a couple sample.json file content samples that _should_ work. The first one exercises the site generation function, and the second one hits the static content (images etc) copy. First try this sample.json

```
{
  "Records": [
    {
      "s3": {
        "object": {
          "eTag": "50ed8c18234b65e3baf1417eac1bb03f",
          "size": 307,
          "key": "content/posts/fake.md"
        },
        "bucket": {
          "arn": "arn:aws:s3:::input.YOURDOMAIN.COM",
          "ownerIdentity": {
            "principalId": "A1LA3Q6S7ANKDV"
          },
          "name": "input.YOURDOMAIN.COM"
        },
        "configurationId": "Kappa-HugoLambdaGenerate-notification",
        "s3SchemaVersion": "1.0"
      },
      "eventVersion": "2.0",
      "eventSource": "aws:s3",
      "awsRegion": "us-east-1",
      "eventTime": "2015-09-08T22:50:04.028Z",
      "eventName": "ObjectCreated:Put",
      "userIdentity": {
        "principalId": "AWS:AIDAJKYN7TTQ6XB6JS7FW"
      },
      "requestParameters": {
        "sourceIPAddress": "64.134.240.129"
      },
      "responseElements": {
        "x-amz-id-2": "l3GSDL5CGzsTAyVnWn9oWaeWuCs+Vic/Z0jmp/uZtfqoqhPiERu3u2GjhbAOQWf23X0qEsKgvzM=",
        "x-amz-request-id": "FEB7EFA6E7254AA4"
      }
    }
  ]
}
```

Then use: 

```
{
  "Records": [
    {
      "s3": {
        "object": {
          "eTag": "50ed8c18234b65e3baf1417eac1bb03f",
          "size": 307,
          "key": "static/fake.css"
        },
        "bucket": {
          "arn": "arn:aws:s3:::input.YOURDOMAIN.COM",
          "ownerIdentity": {
            "principalId": "A1LA3Q6S7ANKDV"
          },
          "name": "input.YOURDOMAIN.COM"
        },
        "configurationId": "Kappa-HugoLambdaGenerate-notification",
        "s3SchemaVersion": "1.0"
      },
      "eventVersion": "2.0",
      "eventSource": "aws:s3",
      "awsRegion": "us-east-1",
      "eventTime": "2015-09-08T22:50:04.028Z",
      "eventName": "ObjectCreated:Put",
      "userIdentity": {
        "principalId": "AWS:AIDAJKYN7TTQ6XB6JS7FW"
      },
      "requestParameters": {
        "sourceIPAddress": "64.134.240.129"
      },
      "responseElements": {
        "x-amz-id-2": "l3GSDL5CGzsTAyVnWn9oWaeWuCs+Vic/Z0jmp/uZtfqoqhPiERu3u2GjhbAOQWf23X0qEsKgvzM=",
        "x-amz-request-id": "FEB7EFA6E7254AA4"
      }
    }
  ]
}
```

**Comment by ryansb** [comment]
Also maybe try redoing "make deps" and "make update"

**Comment by vitr** [comment]
sorry, Amazon must have changed something. This time I started form scratch with the clean repo clone and couldn't run even make create

```
Generated CFN template
aws cloudformation create-stack --stack-name HugoSiteStack --template-body file://hugo-lambda.cfn --capabilities CAPABILITY_IAM

A client error (AlreadyExistsException) occurred when calling the CreateStack operation: Stack [HugoSiteStack] already exists
make: *** [create] Error 255
```

and I don't have any stacks at all.
I gave up((, I'm sure this is something in my aws config, feel free to close the issue.

**Comment by ryansb** [comment]
Hm, you can try doing `aws cloudformation delete-stack --stack-name HugoSiteStack` and recreating the stack. 

Alternatively, if you wait a few days, I'm planning to release v2 of this project which will actually do away with you needing to download anything - you'll be able to do the whole thing from the web console. 

I'm sorry you're having so much trouble getting this working, thanks so much for your patience.

**Comment by vitr** [comment]
thank you for the tip, I believe, aws behaves normally for me today (I could mess up with the regions last time).
I started over again and everything went smoothly, 
just `make create` then sync the demo and the site is up and running in minutes.
I found that this is a hugo issue https://github.com/spf13/hugo/issues/759,
so, no complaints on hug-lambda, great job and I'm definitely looking forward to v2!
